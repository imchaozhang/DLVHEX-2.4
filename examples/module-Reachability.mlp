#module(p1,[]).
v(a).
(e,a,b). 
e(b,c). 
e(c,d).
rc(c) :- @p2[v,e]::reachable(c). 
r(X) :- @p2[v,e]::reachable(X).

#module(p2,[first/1, edge/2]).
edge(a,c1).
edge(c1,c2).
reachable(X) :- first(X).
reachable(Y) :- reachable(X), edge(X,Y).
