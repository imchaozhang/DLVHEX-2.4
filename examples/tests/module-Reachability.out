(p1[{}]={v(a),e(a,b),e(b,c),e(c,d),rc(c),r(a),r(b),r(c),r(d),r(c1),r(c2)}, p2[{first(a),edge(a,b),edge(b,c),edge(c,d)}]={edge(a,c1),edge(c1,c2),first(a),edge(a,b),edge(b,c),edge(c,d),reachable(a),reachable(c1),reachable(c2),reachable(b),reachable(c),reachable(d)})
